presto-native-execution/presto_cpp/main/Announcer.cpp

```diff
@@ -74,7 +74,7 @@ Announcer::Announcer(
     const std::string& address,
     bool useHttps,
     int port,
-    std::function<folly::SocketAddress()> discoveryAddressLookup,
+    const std::shared_ptr<CoordinatorDiscoverer> coordinatorDiscoverer,
```

Comment:
```
const std::shared_ptr<CoordinatorDiscoverer>&

std::shared_ptr<CoordinatorDiscoverer> -> std::move() in ctor intializer
```

Reply:
std::move() has no effect in a const variable. We need a copy anyways. I can change the parameter type to be const &, to reduce one copy. That been said, I think compiler shall do the copy optimization regardless.